lib/Auth/Process/COmanageDbClient.php

```diff
@@ -0,0 +1,480 @@
+<?php
+
+/**
+ * COmanage authproc filter.
+ *
+ * This class is the authproc filter to get attributes from rest api
+ *
+ * Example configuration in the config/config.php
+ *
+ *    authproc.aa = array(
+ *       ...
+ *       '60' => array(
+ *            'class' => 'attrauthcomanage:COmanageDbClient',
+ *            'coId' => 2,
+ *            'userIdAttribute' => 'eduPersonUniqueId',
+ *       ),
+ *
+ * @author Nicolas Liampotis <nliam@grnet.gr>
+ */
+class sspmod_attrauthcomanage_Auth_Process_COmanageDbClient extends SimpleSAML_Auth_ProcessingFilter
+{
+    // List of SP entity IDs that should be excluded from this filter.
+    private $_blacklist = array();
+    
+    private $_coId;
+
+    private $coUserIdType = 'epuid';
+
+    private $_userIdAttribute = 'eduPersonUniqueId';
+
+    // List of VO names that should be included in entitlements.
+    private $voWhitelist = array();
+
+    private $_basicInfoQuery = 'select'
+        . ' person.id,'
+        . ' person.status'
+        . ' from cm_co_people person'
+        . ' inner join cm_co_org_identity_links link'
+        . ' on person.id = link.co_person_id'
+        . ' inner join cm_org_identities org'
+        . ' on link.org_identity_id = org.id'
+        . ' inner join cm_identifiers ident'
+        . ' on org.id = ident.org_identity_id'
+        . ' where'
+        . ' person.co_id = :coId'
+        . ' and not person.deleted'
+        . ' and not link.deleted'
+        . ' and ident.identifier = :coPersonOrgId';
+
+    private $_loginIdQuery = 'select ident.identifier'
+        . ' from cm_identifiers ident'
+        . ' where'
+        . ' ident.co_person_id = :coPersonId'
+        . ' and ident.type = :coPersonIdType'
+        . ' and ident.identifier_id is null';
+
+    private $profileQuery = 'SELECT'
+        . ' name.given, name.family,'
+        . ' mail.mail,'
+        . ' org.affiliation, org.o,'
+        . ' ident.identifier'
+        . ' FROM cm_co_people person'
+        . ' LEFT OUTER JOIN cm_names name'
+        . ' ON person.id = name.co_person_id'
+        . ' LEFT OUTER JOIN cm_email_addresses mail'
+        . ' ON person.id = mail.co_person_id'
+        . ' LEFT OUTER JOIN cm_co_org_identity_links link'
+        . ' ON person.id = link.co_person_id'
+        . ' LEFT OUTER JOIN cm_org_identities org'
+        . ' ON link.org_identity_id = org.id'
+        . ' LEFT OUTER JOIN cm_identifiers ident'
+        . ' ON person.id = ident.co_person_id'
+        . ' WHERE'
+        . ' NOT name.deleted'
+        . ' AND name.name_id IS NULL'
+        . ' AND NOT mail.deleted'
+        . ' AND mail.email_address_id IS NULL'
+        . ' AND NOT link.deleted'
+        . ' AND NOT ident.deleted'
+        . ' AND ident.type = \'uid\''
+        . ' AND ident.identifier_id IS NULL'
```

Comment:
Add checks for affiliation

Reply:
Changes made in "Query improvements" commit